models/src/main/java/com/vimeo/networking2/PermissionActions.kt

```diff
@@ -16,18 +16,18 @@ import com.squareup.moshi.JsonClass
 data class PermissionActions(
 
     @Json(name = "folder.delete_video")
-    val folderDeleteVideo: Boolean? = null,
+    val folderDeleteVideo: Boolean = false,
 
     @Json(name = "folder.edit")
-    val folderEdit: Boolean? = null,
+    val folderEdit: Boolean = false,
 
     @Json(name = "folder.invite")
-    val folderInvite: Boolean? = null,
+    val folderInvite: Boolean = false,
 
     @Json(name = "folder.view")
-    val folderView: Boolean? = null,
+    val folderView: Boolean = false,
 
     @Json(name = "folder.add_subfolders")
-    val folderAddSubFolders: Boolean? = null
+    val folderAddSubFolders: Boolean = false
```

Comment:
I noticed the permissions never return false, only return with a value of null. We'd have to coalesce every permission check with false otherwise, and the nullability of the outer object(PermissionAction) is still there.

Here's a list of example permission actions:

```
    "data": [
        {
            "uri": "/users/79937352/permission_policies/58",
            "name": "Folder Admin",
            "created_on": "2021-03-29T17:27:05+00:00",
            "modified_on": null,
            "permission_actions": {
                "folder.view": true,
                "folder.edit": true,
                "folder.invite": true,
                "folder.delete_video": true,
                "folder.add_subfolders": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/59",
            "name": "Folder Editor",
            "created_on": "2021-03-29T17:27:05+00:00",
            "modified_on": null,
            "permission_actions": {
                "folder.view": true,
                "folder.edit": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/60",
            "name": "Folder Viewer",
            "created_on": "2021-03-29T17:27:05+00:00",
            "modified_on": null,
            "permission_actions": {
                "folder.view": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/2659",
            "name": "Private To Me",
            "created_on": "2021-04-28T19:40:55+00:00",
            "modified_on": null,
            "permission_actions": {
                "folder.view": true,
                "folder.edit": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/2660",
            "name": "Team Denied",
            "created_on": "2021-04-28T19:40:55+00:00",
            "modified_on": null,
            "permission_actions": {
                "folder.view": false,
                "folder.edit": false
            }
        },
        {
            "uri": "/users/79937352/permission_policies/33960",
            "name": "Clip Editor",
            "created_on": "2021-08-24T18:28:24+00:00",
            "modified_on": null,
            "permission_actions": {
                "clip.edit": true,
                "clip.view": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/33961",
            "name": "Clip Editor",
            "created_on": "2021-08-24T18:28:24+00:00",
            "modified_on": null,
            "permission_actions": {
                "clip.edit": true,
                "clip.view": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/33962",
            "name": "Clip Viewer",
            "created_on": "2021-08-24T18:28:24+00:00",
            "modified_on": null,
            "permission_actions": {
                "clip.view": true
            }
        },
        {
            "uri": "/users/79937352/permission_policies/33963",
            "name": "Clip Viewer",
            "created_on": "2021-08-24T18:28:24+00:00",
            "modified_on": null,
            "permission_actions": {
                "clip.view": true
            }
        }
    ]
}
```

Reply:
Based on our conversation yesterday, we established that the API sometimes returns `null` and sometimes returns `false`, as this response shows. I don't recall us coming to any conclusions, but my feeling is that we should honor the potential nullability of these permission actions by not setting default values.